### DISCUSSION TITLE
How to get fields to work with the v3 search api?

### DISCUSSION THREAD

**Discussion by sethchandler** [freelawproject/courtlistener#4392] [Q&A]
I am struggling getting the fields parameter to work when accessing "https://www.courtlistener.com/api/rest/v3/search/". I've tried a variety of ways: using curl, using Python requests. And no matter what I do, I always get **all** the fields for each case back. I'm somewhat new to using courtlistener, so I am hoping it is just a dumb error. What would be most helpful is a couple of examples of successful usings of searches that have various parameters such as court_id or date restrictions AND a fields limitation.

**Comment by mlissner** [comment]
The documentation says: 

![image](https://github.com/user-attachments/assets/cb3d3e13-0665-4d04-b989-de51fe9d22b3)

But that doesn't really clarify that you can't ask for only certain fields to be returned, which I believe is the case. We could spend some time making that work, but it's probably not worth the effort. Better, probably would be to tweak the documentation with this information. I'd love a PR with that tweak, if you're interested.

What I will suggest though, is that if you're building systems now, you might want to check out v4 of the API. We haven't documented it entirely yet, but it should be complete and we'll be rolling it out very soon. Others have been using it for a few months now (documentation and roll out are hard!).
